### ISSUE TITLE
Datepicker not showing days nor months on `ng build` but OK on `ng serve`

### ISSUE THREAD

**Issue by bravier** [valor-software/ng2-bootstrap#4754]
### Versions of ngx-bootstrap, Angular, and Bootstrap:
ngx-bootstrap: 3.1.1
Angular: 6.1.10 / 7.0.1
Bootstrap: 3.3.7
### Build system: Angular CLI

Hello,

Since v3.1.1 the `datepicker` does not seem to work anymore (see attached screenshot) when building our app.
Our app is in French so we are using `defineLocale` to set the lang.
What I tested:
- using `ng serve` it works great ✔
- using `ng build --prod` it produces what you can see on the screenshot ❌

Please feel free to ask for more I'd be happy to help.

![capture d ecran de 2018-10-30 11-33-22](https://user-images.githubusercontent.com/3956758/47723780-8797bf00-dc55-11e8-824e-238f247dbf39.png)

**Comment by Domainv** [comment]
@bravier please provide a code of reproduction for this issue.

**Comment by bravier** [comment]
Thanks for your comment @Domainv.

Unfortunately I can't give you a code that reproduces the issue.
Instead what I can do is to show you how we are using the datepicker.

Here are the involved parts:

_date-chooser.component.ts_
```ts
...
import {defineLocale} from 'ngx-bootstrap/chronos';
import {frLocale} from 'ngx-bootstrap/locale';
...

@Component({
  selector: 'date-chooser',
  templateUrl: './date-chooser.component.html',
    ...
  }],
})
export class DateChooserComponent {
   constructor() {
    defineLocale('fr', frLocale);
    // We also tried using the following (saw in the docs) without success:
    // this.bsLocaleService.use('fr');
  }           
}
```

_date-chooser.component.html_
```html
<p bsDatepicker [bsConfig]='{dateInputFormat: "DD/MM/YYYY", locale: "fr"}' [bsValue]='date'>{{ date | date:'dd/MM/y' }}</p>
```

Hope this helps!

**Comment by ghost** [comment]
I am also facing this same problem, I am on 'en' locale though
<img width="565" alt="screen shot 2018-10-31 at 9 02 19 am" src="https://user-images.githubusercontent.com/42459746/47764182-2bcc4500-dcec-11e8-9ae1-7be8bef49201.png">

**Comment by ghost** [comment]
I don't know If this might help you guys, but I downgraded to 3.0.0 and it's working fine. I suggest downgrading for a while till they fix this problem 👍

**Comment by bravier** [comment]
Absolutely @rabingaire20 we also fixed it by downgrading to 3.0.1.

**Comment by Domainv** [comment]
@bravier This issue related to https://github.com/angular/angular-cli/issues/12112
just set "buildOptimizer": false

**Comment by gowthamkumardoit** [comment]
@Domainv  You save my day 👍

**Comment by sherlock1982** [comment]
@Domainv this issue exists no matter if I compile to es5 or es6. Well I do understand that buildOptimizer causes this but I believe usage of defaut locale could be made more clear to buildOptimizer.

**Comment by sebastian-zarzycki-apzumi** [comment]
It looks to me, that the problem is actually in the code of ngx-bootstrap, rather than in Angular CLI.

**Comment by wtravO** [comment]
Setting the buildOptimizer to false for a single package is not the way to go.  This seems like a library issue with ngx-bootstrap.

**Comment by Domainv** [comment]
closes as duplicate of #4752

**Comment by Domainv** [comment]
@here The issue fixed from 3.1.2 version
